selection-grid-flow/src/main/java/com/vaadin/componentfactory/selectiongrid/SelectionGrid.java

```diff
@@ -194,10 +209,19 @@ private void selectRangeOnly(int fromIndex, int toIndex) {
 
     @Override
     protected void setSelectionModel(GridSelectionModel<T> model, SelectionMode selectionMode) {
-        getElement().executeJs("if (this.querySelector('vaadin-grid-flow-selection-column')) { this.querySelector('vaadin-grid-flow-selection-column').hidden = true }");
+        if (selectionMode == SelectionMode.MULTI) {
+            hideMultiSelectionColumn();
+        }
         super.setSelectionModel(model, selectionMode);
     }
 
+    /**
+     * Runs a JavaScript snippet to hide the multi selection / checkbox column on the client side. The column
+     * is not removed, but set to "hidden" explicitly.
+     */
+    protected void hideMultiSelectionColumn() {
+        getElement().executeJs("if (this.querySelector('vaadin-grid-flow-selection-column')) { this.querySelector('vaadin-grid-flow-selection-column').hidden = true }");
```

Comment:
I think it's better with this:
```
getElement().getNode()
                    .runWhenAttached(ui -> ui.beforeClientResponse(this,
                            context -> getElement().executeJs("if (this.querySelector('vaadin-grid-flow-selection-column')) { this.querySelector('vaadin-grid-flow-selection-column').hidden = true }")));
```

It should also help if you attach/detach the grid in the same round trip